src/client/pydaos/pydaos_core.py

```diff
@@ -392,4 +371,23 @@ def __ne__(self, other):
         return not self.__eq__(other)
 
     def __iter__(self):
-        return KVIter(self)
+        return DDictIter(self)
+
+class DArray(_DObj):
+    """
+    Class representing of DAOS array leveraging the numpy's dispatch mechanism.
+    See https://numpy.org/doc/stable/user/basics.dispatch.html for more info.
+    Work in progress.
+    """
+
+    def _open(self, hdl):
+        raise NotImplemented
```

Comment:
(pylint-notimplemented-raised) NotImplemented raised - should raise NotImplementedError
(pylint-raising-bad-type) Raising NotImplementedType while only classes or instances are allowed